jerry-core/ecma/builtin-objects/ecma-builtin-date.c

```diff
@@ -732,104 +754,107 @@ ecma_value_t
 ecma_builtin_date_dispatch_construct (const ecma_value_t *arguments_list_p, /**< arguments list */
                                       ecma_length_t arguments_list_len) /**< number of arguments */
 {
-  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
-  ecma_number_t prim_value_num = ECMA_NUMBER_ZERO;
-
   ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_DATE_PROTOTYPE);
+
 #if ENABLED (JERRY_ESNEXT)
-  if (JERRY_CONTEXT (current_new_target))
+  JERRY_ASSERT (JERRY_CONTEXT (current_new_target));
+
+  prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
+                                                            ECMA_BUILTIN_ID_DATE_PROTOTYPE);
+  if (JERRY_UNLIKELY (prototype_obj_p == NULL))
   {
-    prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
-                                                              ECMA_BUILTIN_ID_DATE_PROTOTYPE);
-    if (JERRY_UNLIKELY (prototype_obj_p == NULL))
-    {
-      return ECMA_VALUE_ERROR;
-    }
+    return ECMA_VALUE_ERROR;
   }
 #endif /* !(ENABLED (JERRY_ESNEXT) */
+
```

Comment:
How about separate the argument validation into a single function? That would reduce the cost of vale releasing in early returns.